src/main/java/core/basesyntax/service/FruitsServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.FruitsDao;
+import core.basesyntax.service.csvfileservice.ReaderService;
+import core.basesyntax.service.csvfileservice.WriterService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.Map;
+
+public class FruitsServiceImpl implements FruitsService {
+    private FruitsDao fruitsDao;
+    private OperationStrategy operationStrategy;
+    private ReaderService readerService;
+    private WriterService writerService;
+
+    public FruitsServiceImpl(FruitsDao fruitsDao,
+                             OperationStrategy operationStrategy,
+                             ReaderService readerService,
+                             WriterService writerService) {
+        this.fruitsDao = fruitsDao;
+        this.operationStrategy = operationStrategy;
+        this.readerService = readerService;
+        this.writerService = writerService;
+    }
+
+    @Override
+    public void generateFruitsReport() {
```

Comment:
this method should return generated report, call method for writing to file in Main

Reply:
method changed to String return type w/out write method